plugin/nginx_plus_collectd.py

```diff
@@ -772,6 +834,42 @@ def get_processes(self):
         '''
         return self._send_get(self.processes_url)
 
+    def _get_api_version(self):
+        '''
+        Determines whether the Nginx-plus plugin has a versioned API or legacy API.
+        '''
+        newer_api_base_path = self.api_base_path if self.api_base_path is not None else '/api'
+        legacy_api_base_path = self.api_base_path if self.api_base_path is not None else '/status'
+        base_url = 'http://{}:{}'.format(self.status_host, str(self.status_port))
+
+        try:
+            response = requests.get("{}{}/{}".format(base_url, newer_api_base_path, DEFAULT_API_VERSION), auth=self.auth_tuple)
+            if response.status_code == requests.codes.ok:
+                return DEFAULT_API_VERSION
+            else:
+                response = requests.get("{}{}".format(base_url, legacy_api_base_path), auth=self.auth_tuple)
+                if response.status_code == requests.codes.ok:
+                    return 0
+            LOGGER.error(
```

Comment:
Instead of logging errors and returning None, why not raise an exception that the caller logs and accounts for?

Reply:
Makes sense. We have made the change accordingly.